### PR TITLE
Added teflo_plugin_template dir

### PR DESCRIPTION
**Pull request by rujutashinde**
* This dir is a template for creating teflo plugins

### CODE REVIEW

File: teflo_plugin_template/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    <td class="tg-8m83">test_provx_plugin</td>
+  </tr>
+  <tr>
+    <td class="tg-8m83">plugin_description</td>
+    <td class="tg-8m83">Plugins description that goes into the setup.py</td>
+    <td class="tg-8m83">teflo provisioner plugin</td>
+  </tr>
+  <tr>
+    <td class="tg-8m83">jenkins_ci_job_link</td>
+    <td class="tg-8m83">jenkins ci job link once you have created that. This gets  updated in the jenkins/Jenkinsfile</td>
```
Comment (shay6) [inline teflo_plugin_template/README.md]: unnecessary space before "updated"

File: teflo_plugin_template/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    <td class="tg-8m83">your ci job link</td>
+  </tr>
+  <tr>
+    <td class="tg-8m83">plugin_url</td>
+    <td class="tg-8m83">plugin url needed to start the ci job. This gets  updated in the jenkins/Jenkinsfile</td>
```
Comment (shay6) [inline teflo_plugin_template/README.md]: unnecessary space

File: teflo_plugin_template/README.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+###  Example
+``` bash
+$ cookiecutter <path to the teflo_examples dir>/teflo_plugin_template/
+teflo_plugin_type [provisioner]: importer
+directory_name [teflo_provisionerX_plugin]: teflo_datarouter_plugin
+plugin_name [provx_plugin]: datarouter_plugin
+plugin_class_name [ProvXProvisionerPlugin]: DataRouterImporterPlugin
+test_class_name [test_datarouter_plugin]: 
+plugin_description [teflo provisioner plugin]: teflo pluging for data router
```
Comment (shay6) [inline teflo_plugin_template/README.md]: plugin instead of pluging

**Review by shay6 — COMMENTED** [review]
LGTM

File: teflo_plugin_template/README.md (hunk 1 above)
Comment (rujutashinde) [reply teflo_plugin_template/README.md]: done

File: teflo_plugin_template/README.md (hunk 2 above)
Comment (rujutashinde) [reply teflo_plugin_template/README.md]: done

File: teflo_plugin_template/README.md (hunk 3 above)
Comment (rujutashinde) [reply teflo_plugin_template/README.md]: done

**Comment by shay6** [comment]
I downloaded the `template` branch, I tried out the cookiecutter command with the example you wrote and it works very good.
LGTM

**Review by JunqiZhang0 — APPROVED** [review]
LGTM

**Pull request merged**